### DISCUSSION TITLE
Callbook Search - No contact info populated from QRZ (or Hamqth)

### DISCUSSION THREAD

**Discussion by faocrates** [magicbug/Cloudlog#3043] [General]
I have configured qrz as my callbook in config.php using the following,

$config['callbook'] = "qrz"; // Options are hamqth or qrz
$config['qrz_username'] = "<callsign_username>";
$config['qrz_password'] = "<my_password>";

Still when I type a callsign and press Tab to change field, not contact info is populated. The message appears: (Callbook Search for <Callsign>) "Lookup not configured. Please review configuration."

The same problem appears when I configure hamqth instread,

$config['callbook'] = "hamqth"; // Options are hamqth or qrz
$config['hamqth_username'] = "<callsign_username>";
$config['hamqth_password'] = "<my_password>";

Am I missing something? Any help is greatly appreciated.

**Comment by faocrates** [comment]
Update from  me, I noticed the original issue, the callbook config appears within config.php in an 'Install Options' section and in 'QRZ Login Options', I had set values in the first section which were overwritten with "" in the second. I've corrected this and now the 'Lookup not configured' message is gone. Still though in e.g. 'Post QSO' when I enter a callsign and leave the field with a tab no contact info is auto-filled. Any help with this? Is this a configuration option within Cloudlog or do I need to configure something within QRZ?

**Comment by magicbug** [comment]
for QRZ to work you should just need to populate

$config['callbook'] = "qrz"; // Options are hamqth or qrz
$config['qrz_username'] = "<callsign_username>";
$config['qrz_password'] = "<my_password>";

in config.php, when you then type a callsign in it will look it up on QRZ. if this doesn't work then there's a server config issue in config.php enable all logging output and then check application/logs folder and see whats being reported

**Reply by faocrates** [reply]
Thank you for your reply. How do I enable all log output? Since installation, no log files have been created in the application/logs folder. There is just an index.html there reporting 403 Forbidden - Directory access is forbidden. The logs folder has all ugo+rwx permissions already. Any suggestion?

**Reply by Hixsten** [reply]
Hi,
 look for log_threshold in application/config/config.php file.

**Reply by R1BLH** [reply]
Below is the log file that is created when requesting data from qrz.com when filling out the new QSO form, if the callsign is on qrz.com and previous QSOs with this station are not recorded in the log.
In this case there is no request for data from qrz.com. 
Everything works fine with hamqth.com.

How can I diagnose and correct this error?

> <?php defined('BASEPATH') OR exit('No direct script access allowed'); ?>
> 
> DEBUG - 2024-04-15 00:36:48 --> No URI present. Default controller set.
> ERROR - 2024-04-15 00:36:49 --> Could not find the language line "menu_view_sstv"
> ERROR - 2024-04-15 00:36:51 --> Could not find the language line "menu_view_sstv"
> DEBUG - 2024-04-15 00:36:52 --> Frequency class already loaded. Second attempt ignored.
> DEBUG - 2024-04-15 00:36:58 --> Qrz class already loaded. Second attempt ignored.
> ERROR - 2024-04-15 00:37:13 --> Severity: error --> Exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 1 /var/www/r1blh.cats-home.net/log/system/database/drivers/mysqli/mysqli_driver.php 301
>

**Comment by magicbug** [comment]
I'd suggest checking against dev branch as I believe it's fixed

**Reply by faocrates** [reply]
Thank you for confirming on a fix, any idea when this fix may be released from master?

**Reply by R1BLH** [reply]
Just updated to the latest dev branch. 
![изображение](https://github.com/magicbug/Cloudlog/assets/16667173/1fcf15db-116c-44f9-85f5-d8b8454b67e1)

Situation with qrz lookup has not changed.

> DEBUG - 2024-04-19 03:46:46 --> Frequency class already loaded. Second attempt ignored.
> ERROR - 2024-04-19 03:47:05 --> Severity: error --> Exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '' at line 1 /var/www/r1blh.cats-home.net/log/system/database/drivers/mysqli/mysqli_driver.php 301
>

**Comment by stevendodd** [comment]
I was really struggling with this as well, the solution for me was to use my call sign as the username in both the config.php and the account settings in the application. Until I set this consistently in both places it wouldn't work.

**Comment by magicbug** [comment]
To my knowledge, the config file variables aren't called anywhere in Cloudlog anymore if anyone can find any references they will be sorted soon as possible
